app/javascript/components/pages/Show.js

```diff
@@ -1,9 +1,103 @@
 import React, { Component } from 'react'
+import { Card, CardImg, CardImgOverlay, CardTitle, CardText, Button, ListGroup, ListGroupItem, ListGroupItemHeading, ListGroupItemText } from 'reactstrap'
+import { NavLink, Link } from 'react-router-dom'
+import reviews from '../MockReviews';
+import CreateReview from './CreateReview'
+
+
+
 
 export default class Show extends Component {
+  constructor(props) {
+    super(props)
+    console.log(props, "props")
+    this.state = {
+      listingID: this.props.listing.id,
+      review: false
+    }
+  }
+
+  handleClick = (e) => {
+    e.preventDefault()
+    this.setState({review: !this.state.review})
+  }
+
+
   render() {
+
+    const {
+      logged_in,
+      current_user,
+      new_user_route,
+      sign_in_route,
+      sign_out_route,
+    } = this.props;
+
+
+    let { review } = this.props
+    let { listing } = this.props
```

Comment:
Since all of these are being destructured from this.props, this can be consolidated.